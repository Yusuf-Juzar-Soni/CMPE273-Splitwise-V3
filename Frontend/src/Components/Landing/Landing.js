import React from "react";
import Navbar from "react-bootstrap/Navbar";
import Button from "react-bootstrap/Button";
import { Nav } from "react-bootstrap";
import bg_image0 from "../assets/login_logo.png";
import bg_image1 from "../assets/splitwise_home_2.png";
import bg_image2 from "../assets/splitwise_home_3.png";

const Landing = function () {
  return (
    <div>
      <div>
        <Navbar bg="success" expand="lg">
          <Navbar.Brand>
            <img src={bg_image0} width="50" height="50"></img>
          </Navbar.Brand>
          <Nav className="mr-auto">
            <Nav.Link href="#home" />
          </Nav>
          <Button data-testid="Login" variant="success" href="./login">
            Login
          </Button>
          <Button variant="success" href="./signup">
            Signup
          </Button>
        </Navbar>
      </div>
      <div>
        <img src={bg_image1} width="1350" height="600"></img>
        <img
          src={bg_image2}
          width="1350"
          height="600"
          style={{ marginTop: 80 }}
        ></img>
      </div>
    </div>
  );
};

export default Landing;
