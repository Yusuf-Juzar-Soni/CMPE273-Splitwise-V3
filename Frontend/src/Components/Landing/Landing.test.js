import React from "react";
import { render, cleanup } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { createStore } from "redux";
import { MemoryRouter } from "react-router-dom";
// import configureStore from 'redux-mock-store';
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import Landing from "./Landing";
import combReducers from "../../reducers";

const store = createStore(combReducers);

const component = TestRenderer.create(
  <Provider store={store}>
    <MemoryRouter>
      <Landing></Landing>
    </MemoryRouter>
  </Provider>
);

afterEach(cleanup);

it("renders", async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test("Check for header", async () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Landing></Landing>
      </MemoryRouter>
    </Provider>
  );
  expect(getByTestId("Login")).toHaveTextContent("Login");
});
