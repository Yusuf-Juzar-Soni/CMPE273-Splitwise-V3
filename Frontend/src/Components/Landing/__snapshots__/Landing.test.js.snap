// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light bg-success"
    >
      <span
        className="navbar-brand"
      >
        <img
          height="50"
          src="login_logo.png"
          width="50"
        />
      </span>
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="#home"
          href="#home"
          onClick={[Function]}
          onKeyDown={[Function]}
        />
      </div>
      <a
        className="btn btn-success"
        data-testid="Login"
        href="./login"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Login
      </a>
      <a
        className="btn btn-success"
        href="./signup"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Signup
      </a>
    </nav>
  </div>
  <div>
    <img
      height="600"
      src="splitwise_home_2.png"
      width="1350"
    />
    <img
      height="600"
      src="splitwise_home_3.png"
      style={
        Object {
          "marginTop": 80,
        }
      }
      width="1350"
    />
  </div>
</div>
`;
